### PR TITLE
Migrate to Nextclade v3

### PR DESCRIPTION
**Pull request by joverlee521**
## Description of proposed changes

Main goal is to migrate to Nextclade v3. I discovered Nextclade related params and rules that were unused, so this PR also removes them to make the migration easier. 

See commits for details. 

## Related issue(s)

Resolves https://github.com/nextstrain/ncov/issues/1157

## Release checklist

If this pull request introduces backward incompatible changes, complete the following steps for a new release of the workflow:

 - [x] Determine the version number for the new release by incrementing [the most recent release](https://github.com/nextstrain/ncov/releases) (e.g., "v2" from "v1").
 - [x] Update `docs/src/reference/change_log.md` in this pull request to document these changes and the new version number.
 - [ ] After merging, [create a new GitHub release](https://github.com/nextstrain/ncov/releases/new) with the new version number as the tag and release title.

### CODE REVIEW

File: workflow/snakemake_rules/main_workflow.smk [hunk 1]
```diff
@@ -503,13 +493,12 @@ rule build_align:
...
             --jobs {threads} \
             --input-dataset {input.nextclade_dataset} \
             --output-tsv {output.nextclade_qc} \
             --output-fasta {output.alignment} \
-            --output-translations {params.output_translations} \
-            --output-insertions {output.insertions} 2>&1 | tee {log}
+            --output-translations {params.output_translations} 2>&1 | tee {log}
         """
 
 rule join_metadata_and_nextclade_qc:
```
Comment (joverlee521) [inline workflow/snakemake_rules/main_workflow.smk:504]: I was tempted to update this rule to use `augur merge` to join the metadata and Nextclade data. However, the custom `join-metadata-and-clades.py` script includes [calculations for `clock_deviation`](https://github.com/nextstrain/ncov/blob/66aa9be182df05f04acc5a2cbb4ad750b39eb0ec/scripts/join-metadata-and-clades.py#L125-L143) that I don't really want to touch on here. Will leave that for another time...

**Comment by joverlee521** [comment]
> Emma used mutation summary in covariants back in the day. I am not sure if it is still used. If file upload was removed, then probably not. Or she could be running it outside of the official AWS-based flow.

Thanks for the reminder! I found an old [Slack thread](https://bedfordlab.slack.com/archives/C01LCTT7JNN/p1647538476963439?thread_ts=1647353315.650509&cid=C01LCTT7JNN) that confirms @emmahodcroft no longer used the mutation summary produced by ncov-ingest, but I could not find anything regarding the mutation summary produced here. 

> Without annotation it might produce lower quality alignments and change what mutations are detected, so I think it's better to keep annotations, even if translation is not being run. If so decided, then might also add a comment or a description that even though annotation appears unused, it is in fact used for codon-aware nucleotide alignment (and perhaps a link to the docs):
> 
> https://docs.nextstrain.org/projects/nextclade/en/stable/user/algorithm/01-sequence-alignment.html

Thanks for the pointer! That's good to know. I'll add back annotations.

File: docs/src/reference/workflow-config-file.rst [hunk 2]
```diff
@@ -513,17 +513,17 @@ alignment_reference
...
 -  default: ``defaults/reference_seq.fasta``
--  used in rules: ``align``, ``proximity_score`` (subsampling), ``build_align``, ``build_mutation_summary``
+-  used in rules: ``align``, ``proximity_score`` (subsampling)
 
 annotation
 ~~~~~~~~~~
 
 -  type: string
--  description: Path to a GFF-formatted annotation of gene coordinates (e.g., a “gene map”) for use by ``nextalign`` and mutation summaries.
+-  description: Path to a GFF-formatted annotation of gene coordinates (e.g., a “gene map”) for use by ``nextalign`` for codon-aware alignment.
```
Comment (genehack) [inline docs/src/reference/workflow-config-file.rst:524]: ?? 

```suggestion
-  description: Path to a GFF-formatted annotation of gene coordinates (e.g., a “gene map”) for use by ``nextclade`` for codon-aware alignment.
```

**Review by genehack — APPROVED** [review]

File: docs/src/reference/workflow-config-file.rst (hunk 2 above)
Comment (joverlee521) [reply docs/src/reference/workflow-config-file.rst:524]: Ah, good catch! I missed this during rebasing.

Comment (joverlee521) [reply docs/src/reference/workflow-config-file.rst]: Fixed in [force-push](https://github.com/nextstrain/ncov/compare/c5ca8fee2c60a14786d51716ed90e8795bd536c9..f269e69b4921f93c37a0964accd4bdfc418ea1c5)

**Pull request merged**